File: sgifd-frontend/src/app/models/SousSecteur.ts
import {Secteur} from "./Secteur";

export class SousSecteur {
  id: number;
  libelle:	string;
  createBy:	string;
  deleteBy:	string;
  secteur: Secteur;

  constructor(id: number, libelle: string, createBy: string, deleteBy: string, secteur: Secteur) {
    this.id = id;
    this.libelle = libelle;
    this.createBy = createBy;
    this.deleteBy = deleteBy;
    this.secteur = secteur;
  }
}

File: sgifd-backend/src/main/java/com/sgifdbackend/demo/parametrage/entites/SousSecteur.java
package com.sgifdbackend.demo.parametrage.entites;

import com.sgifdbackend.demo.entitiesBase.EntityBaseBean;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "t_SousSecteur")
@EntityListeners(AuditingEntityListener.class)
public class SousSecteur extends EntityBaseBean implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String libelle;

    @ManyToOne
    private Secteur secteur;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getLibelle() {
		return libelle;
	}

	public void setLibelle(String libelle) {
		this.libelle = libelle;
	}

	public Secteur getSecteur() {
		return secteur;
	}

	public void setSecteur(Secteur secteur) {
		this.secteur = secteur;
	}

    
}
